contents/python/introduction-to-python.md

```diff
@@ -4,25 +4,30 @@ Python is a simple yet powerful and versatile language. Conceived in the late 80
 
-[Google's Python Class](https://developers.google.com/edu/python/)
+If you're reading this on GitHub, I'm guessing that you're already an experienced programmer looking to get in on the Python action. If that's the case, check out [Google's Python class](https://developers.google.com/edu/python/), which will introduce you to Python's concepts and core data structures like lists, dictionaries, and strings!
+
+For absolute beginners, check out [this video](https://www.youtube.com/watch?v=N4mEzFDjqtA) by Derek Banas where he covers everything from installing Python and the basics to Inheritance and Polymorphism(!!!) in under an hour! If you'd prefer to read, check out [Python Guru](http://thepythonguru.com/) which has plenty of code samples to help you along.
+
+Both newbies and experienced programmers can also benefit from [The Python Tutorial](https://docs.python.org/3/tutorial/index.html), which aims to introduce readers to Python's unique features and style.
 
 ## Iterables
 
-[Iterables, Iterators and Generators](https://excess.org/article/2013/02/itergen1/)
+Strings, lists, and dictionaries belong to a type of class known as `Iterables`. If you're interested in delving nto the inner workings of `Iterable` classes, check out [Iterables, Iterators and Generators](https://excess.org/article/2013/02/itergen1/) by Ian Ward. He begins by explaining the `Iterable` class, then goes into the `Iterator` and `Generator` classes, both of which are powerful tools in Python.
 
-## Slicing
+## Indexing and Slicing
 
-[Introduction to indexing and slicing](https://samsontmr.github.io/Slicing-and-Dicing/)
+Accessing an element in a list using its index is known as indexing, e.g. `my_list[0]` returns the first element of my_list (Iterables are zero-indexed; this means that the first index is 0). Slicing, on the other hand, allows us to access a range of elements in a list. Extended slicing extends this functionality by introducing a step parameter, allowing us to do things like access every other element. Read more about this inn my blog post on [indexing and slicing](https://samsontmr.github.io/Slicing-and-Dicing/) in Python,
 
-## OOP in Python
+## Object Oriented Programming in Python
 
-[Python Class and OOP](https://jeffknupp.com/blog/2014/06/18/improve-your-python-python-classes-and-object-oriented-programming/)
+While Python is really useful for writing small scripts, often we want to use it to write whole applications. When this happens, having everything in one script (we call this procedural programming) will create huge problems readability-wise, especially when things start breaking and you're trying to find out why! This is where Object Oriented Programming comes in, offering a way to think about your program and organizing it into readable chunks. Jeff Knupp gives an introduction to [Python Classes and OOP](https://jeffknupp.com/blog/2014/06/18/improve-your-python-python-classes-and-object-oriented-programming/) here.
```

Comment:
I think you should also state the more important aspects of OOP like Polymorphisms, Overloading, Overriding, Inheritance, so that the reader knows what he/she should take away when learning about OOP. Besides, you already made a reference to Polymorphism and Inheritance above, can reiterate here again